lib/screens/auth/screens/reset_password_screen.dart

```diff
@@ -0,0 +1,231 @@
+                    child: Column(
+                      children: [
+                        Padding(
+                          padding:
+                              const EdgeInsets.only(left: 40.0, right: 40.0),
+                          child: RichText(
+                            textAlign: TextAlign.center,
+                            text: TextSpan(
+                              children: [
+                                TextSpan(
+                                  text: 'Reset Password\n\n',
+                                  style: TextStyle(
+                                      color: kWhiteTextColor, fontSize: 38),
+                                ),
+                                TextSpan(
+                                  text:
+                                      'Please enter your email address. You will receive a link to create a new password via email',
+                                  style: TextStyle(
+                                      // color: Colors.grey.shade500,
+                                      fontSize: 16),
+                                ),
+                              ],
+                            ),
+                          ),
+                        ),
+                        SizedBox(height: 18.0),
+                        Container(
+                          margin: EdgeInsets.only(left: 40, right: 40),
+                          child: LoginField(
+                            key: Key('email-field'),
+                            controller: _emailField,
+                            validator: (value) => (value.isEmpty)
+                                ? 'Please enter a valid email'
+                                : null,
+                            hintText: 'enter your email',
+                            labelText: 'E-mail',
+                            isPassword: false,
+                          ),
+                        ),
+                        SizedBox(height: 18.0),
+                      ],
+                    ),
+                  ),
+                ),
+                SizedBox(height: 20),
+                ReusableButton(
+                  onPressed: () async {
+                    await _resetPassword();
+                  },
+                  text: 'SEND',
+                ),
+              ],
+            ),
+          ),
+        ));
+  }
+
+  _resetPassword() async {
+    final user = Provider.of<UserRepository>(context, listen: false);
+    print(_emailField.text);
+    if (_formKey.currentState.validate()) {
+      await user
+          .resetPassword(_emailField.text)
+          .then((value) => _showResetPasswordDialog(context));
+    }
+  }
+
+  _showResetPasswordDialog(BuildContext context) async {
+    final user = Provider.of<UserRepository>(context, listen: false);
+    if (!await user.resetPassword(_emailField.text)) {
+      return showDialog(
+          context: context,
+          builder: (context) {
+            return AlertDialog(
```

Comment:
Use here the same strategy with ternary operators. On both parts of the if/else, you are returning the same widget function (showDialog) with just a different logic insice.

Reply:
I don't know how to use the ternary operators when I have a condition with "await"